daemon/client_test.go

```diff
@@ -0,0 +1,363 @@
+	expectedProcResponse := ""
+	procName := "run-sample"
+	procArgs := map[string]string{"SAMPLE_ARG1": "sample-value"}
+	err := ioutil.WriteFile(proctorConfigFilePath, proctorConfig, 0644)
+	assert.NoError(t, err)
+	config.InitConfig()
+
+	proctorClient := NewClient()
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterStubRequest(
+		httpmock.NewStubRequest(
+			"POST",
+			"http://"+config.ProctorURL()+"/jobs/execute",
+			func(req *http.Request) (*http.Response, error) {
+				return httpmock.NewStringResponse(500, ""), nil
+			},
+		).WithHeader(
+			&http.Header{
+				"Email-Id":     []string{"proctor@example.com"},
+				"Access-Token": []string{"access-token"},
+			},
+		),
+	)
+
+	executeProcResponse, err := proctorClient.ExecuteProc(procName, procArgs)
+
+	assert.Nil(t, err)
+	assert.Equal(t, expectedProcResponse, executeProcResponse)
+
+	if proctorConfigFileExistedBeforeTest {
+		err = ioutil.WriteFile(proctorConfigFilePath, existingConfigFileData, 0644)
+		assert.NoError(t, err)
+	} else {
+		err = os.Remove(proctorConfigFilePath)
+		assert.NoError(t, err)
+
+		err := os.Remove(proctorConfigDir)
+		assert.NoError(t, err)
+	}
+}
+
+func TestExecuteProcUnAuthorized(t *testing.T) {
+	proctorConfigFileExistedBeforeTest, proctorConfigFilePath, existingConfigFileData, proctorConfigDir := readProctorConfigFile()
+	proctorConfig := []byte("PROCTOR_URL: proctor.example.com\nACCESS_TOKEN: access-token\nEMAIL_ID: proctor@example.com")
+	expectedProcResponse := ""
+	procName := "run-sample"
+	procArgs := map[string]string{"SAMPLE_ARG1": "sample-value"}
+	err := ioutil.WriteFile(proctorConfigFilePath, proctorConfig, 0644)
+	assert.NoError(t, err)
+	config.InitConfig()
+
+	proctorClient := NewClient()
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterStubRequest(
+		httpmock.NewStubRequest(
+			"POST",
+			"http://"+config.ProctorURL()+"/jobs/execute",
+			func(req *http.Request) (*http.Response, error) {
+				return httpmock.NewStringResponse(401, ""), nil
+			},
+		).WithHeader(
+			&http.Header{
+				"Email-Id":     []string{"proctor@example.com"},
+				"Access-Token": []string{"access-token"},
+			},
+		),
+	)
+
+	executeProcResponse, err := proctorClient.ExecuteProc(procName, procArgs)
+
+	assert.Equal(t, expectedProcResponse, executeProcResponse)
+	assert.Equal(t, err.Error(), "Unauthorized")
+
+	if proctorConfigFileExistedBeforeTest {
+		err = ioutil.WriteFile(proctorConfigFilePath, existingConfigFileData, 0644)
+		assert.NoError(t, err)
+	} else {
+		err = os.Remove(proctorConfigFilePath)
+		assert.NoError(t, err)
+
+		err := os.Remove(proctorConfigDir)
+		assert.NoError(t, err)
+	}
+}
+
+func TestLogStream(t *testing.T) {
+	testServer := httptest.NewServer(Stream())
+	defer testServer.Close()
+
+	proctorConfigFileExistedBeforeTest, proctorConfigFilePath, existingConfigFileData, proctorConfigDir := readProctorConfigFile()
```

Comment:
Because config isn't a struct, you have to do this complex thing of writing to the config file everytime you write tests requiring config.

A way to simplify this is making `config` a struct and injecting it as dependency to proctord client.

Your config interface will look like this:
```
type Config interface {
	ProctorURL() (string, error)
	EmailId() (string, error)
	AccessToken() (string, error)
}
```
Your proctord Client will look like this:
```
func NewClient(config config.Config) Client {
return &client{
		proctorEngineURL: config.ProctorURL(),
		userEmailId: config.EmailId(),
                userAccessToken: config.AccessToken(),
	}
```
Post this, you can mock the methods on config object in your test to return custom URL/email-ID/access token as required by the test.